### DISCUSSION TITLE
How to set start and end points of an arc.

### DISCUSSION THREAD

**Discussion by xDylan03x** [mozman/ezdxf#714] [Q&A] [answered]
I am trying to create a circular segment (or arc segment) and it seems that the 'start_point' and 'end_point' parameters are no longer accepted. What would be the best way to go about this? Thanks

`arc = document.msp.add_arc((self.midPoint, -38.3214), 38.3214, 0, 180, start_point=self.startPoint, end_point=self.endPoint, dxfattribs={"layer": "Visible"})`

`TypeError: CreatorInterface.add_arc() got an unexpected keyword argument 'start_point'`

**Comment by mozman** [comment, marked as answer]
The `add_arc()` method had never the arguments `start_point` and `end_point` because it always used only the geometry parameters `center`, `radius`, `start_angle` and `end_angle` of the underlying DXF `ARC` entity to create an arc.

You can use the construction tool [ezdxf.math.ConstructionArc](https://ezdxf.mozman.at/docs/math/core.html#constructionarc) to create arcs from 3 points by the classmethod [from_3p()](https://ezdxf.mozman.at/docs/math/core.html#ezdxf.math.ConstructionArc.from_3p) and add the arc by the [add_to_layout()](https://ezdxf.mozman.at/docs/math/core.html#ezdxf.math.ConstructionArc.add_to_layout) method to a layout like the modelspace.

```Python
import ezdxf
from ezdxf.math import ConstructionArc

doc = ezdxf.new()
arc = ConstructionArc.from_3p(start_point=(0, 0), end_point=(10, 0), def_point=(5, 3), ccw=False)
arc.add_to_layout(doc.modelspace(), dxfattribs={"color": ezdxf.colors.RED})
doc.saveas("arc.dxf")
```
![image](https://user-images.githubusercontent.com/4230966/182276823-27c292d1-4f20-4f0c-ad62-49e74a973b05.png)

**Reply by xDylan03x** [reply]
That worked great. Thanks.
